root/static/scripts/common/components/ExpandedArtistCredit.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * @flow
+ * This file is part of MusicBrainz, the open internet music database.
+ * Copyright (C) 2019 MetaBrainz Foundation
+ * Licensed under the GPL version 2, or (at your option) any later version:
+ * http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import commaOnlyList from '../i18n/commaOnlyList';
+import {reduceArtistCredit} from '../immutable-entities';
+
+import ArtistCreditLink from './ArtistCreditLink';
+import DescriptiveLink from './DescriptiveLink';
+
+export const ExpandedArtistCreditList = ({
+  artistCredit,
+}: {artistCredit: ArtistCreditT}) => {
+  if (!artistCredit) {
+    return null;
+  }
+
+  const names = artistCredit.names;
+  const artistList = [];
+  let showList = false;
+
+  if (reduceArtistCredit(artistCredit) !== names[0].artist.name ||
+    names.length > 1 || names[0].artist.comment) {
+    names.map(name => {
```

Comment:
You can assign the result of the map here to `artistList = ` and change `artistList.push` in the map callback to just `return`. The map will return a list of React elements then.